pkg/backend/display/object_diff.go

```diff
@@ -1215,13 +1220,23 @@ func (p *propertyPrinter) decodeValue(repr string) (resource.PropertyValue, stri
 			return nil, "", false
 		}
 
-		r.Reset(repr)
-		if err := yaml.NewDecoder(r).Decode(&object); err == nil {
-			translated, ok := p.translateYAMLValue(object)
-			if !ok {
-				return nil, "", false
+		pos, err := r.Seek(0, io.SeekStart)
+		contract.Assertf(err == nil, "strings.NewReader.Seek(0, io.SeekStart) failed: %v", err)
+		contract.Assertf(pos == 0, "strings.NewReader.Seek(0, io.SeekStart) failed: pos=%d", pos)
```

Comment:
nit: I feel like trimming repr at the start, and then just `Reset(repr)` here might be more readable, and would allow dropping the impossible error case here?

```go
decode := func(...) {
  repr := strings.TrimSpace(repr) // := so it doesn't change the outer value
  // ...
  r.Reset(repr)
}
```

Reply:
Sure